assets/js/sharedClientFunctions.js

```diff
@@ -80,11 +81,10 @@ module.exports = {
       });
     };
     $scope.deleteRef = function (id) {
-      var current_user = $scope.refUser || $scope.user;
       var url = "/reference/delete";
       io.socket.post(url, {refId: id}, function (data, res) {
         if (res.statusCode === 200) {
-          current_user.references = current_user.references.filter(function (ref) {
+          $scope[current_user].references = $scope[current_user].references.filter(function (ref) {
```

Comment:
I still don't understand how it doesn't work, because refUser should still only exist when refCtrl is using it.